### PR TITLE
Feature nonnupkg

### PR DESCRIPTION
**Pull request by zentron**
`LightweightPackageExtractor` is now `OpenPackagingConventionExtractor` which is utilized through the `GenericPackageExtractor` whos job it is to defer the extraction to the appropriate class depending on format.

To support work that needs to be done on the server, the existing packages file extensions will be included in the `FindPackage` command.

Connects to OctopusDeploy/Issues#2183

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -133,6 +133,8 @@ publish/
 
 # NuGet Packages
 *.nupkg
+!**/*/Acme.Core.1.0.0.0-bugfix.nupkg
```
Comment (PaulStovell) [inline .gitignore]: Seems awfully specific :)

File: source/Calamari.Tests/Fixtures/Integration/Packages/GenericPackageExtractorFixture.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+        [Test]
+        [ExpectedException(typeof(FileFormatException), ExpectedMessage = "Package is missing file extension. This is needed to select the correct extraction algorithm.")]
+        public void FileWithNoExtensionThrowsError()
+        {
+            extractor.GetExtractor("blah");
+        }
+
+
```
Comment (PaulStovell) [inline source/Calamari.Tests/Fixtures/Integration/Packages/GenericPackageExtractorFixture.cs]: Too much whitespace! :)

File: source/Calamari/Integration/Packages/SimplePackageExtractor.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            var metaDataSection = string.Empty;
+            if (matchingExtension != null)
+            {
+                metaDataSection = fileName.Substring(0, fileName.Length - matchingExtension.Length);
+            }
+            else
+            {
+                foreach (var ext in Extensions)
+                {
+                    var match = new Regex("(?<extension>" + ext.Replace(".", "\\.") + ")-[a-z0-9\\-]*$").Match(fileName);
```
Comment (PaulStovell) [inline source/Calamari/Integration/Packages/SimplePackageExtractor.cs]: I think `Regex.Escape` might be better than `.Replace(".",...)` -- but then maybe string.IndexOf() >= 0 might work better too?

File: .gitignore (hunk 1 above)
Comment (zentron) [inline .gitignore]: :) Just a resource used for a test, didn't see any reason to go more generic.

**Pull request closed without merging**